src/Debugger/Impl/DataInspection/REvaluationInfo.cs

```diff
@@ -0,0 +1,104 @@
+                return new RPromiseInfo(session, environmentExpression, expression, name, code);
+            }
+
+            var isActiveBinding = json.Value<bool?>("active_binding");
+            if (isActiveBinding == true) {
+                return new RActiveBindingInfo(session, environmentExpression, expression, name);
+            }
+
+            return new RValueInfo(session, environmentExpression, expression, name, json);
+        }
+
+        public Task SetValueAsync(string value, CancellationToken cancellationToken = default(CancellationToken)) {
+            if (string.IsNullOrEmpty(Expression)) {
+                throw new InvalidOperationException(Invariant($"{nameof(SetValueAsync)} is not supported for this {nameof(REvaluationInfo)} because it doesn't have an associated {nameof(Expression)}."));
+            }
+            return Session.ExecuteAsync($"{Expression} <- {value}", REvaluationKind.Mutating, cancellationToken);
+        }
+
+        public async Task<IReadOnlyList<IREvaluationInfo>> DescribeChildrenAsync(
+            RValueProperties fields,
+            int? maxCount = null,
+            string repr = null,
+            CancellationToken cancellationToken = default(CancellationToken)
+        ) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            if (EnvironmentExpression == null) {
+                throw new InvalidOperationException("Cannot retrieve children of an evaluation result that does not have an associated environment expression.");
+            }
+            if (Expression == null) {
+                throw new InvalidOperationException("Cannot retrieve children of an evaluation result that does not have an associated expression.");
+            }
+
+            var call = Invariant($"rtvs:::describe_children({Expression.ToRStringLiteral()}, {EnvironmentExpression}, {fields.ToRVector()}, {maxCount}, {repr})");
+            var jChildren = await Session.EvaluateAsync<JArray>(call, REvaluationKind.Normal, cancellationToken);
+            Trace.Assert(
+                jChildren.Children().All(t => t is JObject),
+                Invariant($"rtvs:::describe_children(): object of objects expected.\n\n{jChildren}"));
+
+            var children = new List<REvaluationInfo>();
+            foreach (var child in jChildren) {
+                var childObject = (JObject)child;
+                Trace.Assert(
+                    childObject.Count == 1,
+                    Invariant($"rtvs:::describe_children(): each object is expected contain one object\n\n"));
+                foreach (var kv in childObject) {
+                    var name = kv.Key;
+                    var jEvalResult = (JObject)kv.Value;
+                    var evalResult = Parse(Session, EnvironmentExpression, name, jEvalResult);
+                    children.Add(evalResult);
+                }
+            }
+
+            return children;
+        }
+
+        public override bool Equals(object obj) =>
```

Comment:
This is leftover from when the tests were using match objects. I'll get rid of it.

---
In reply to: [62135945](https://github.com/Microsoft/RTVS/pull/1731#discussion_r62135945) [](ancestors = 62135945)